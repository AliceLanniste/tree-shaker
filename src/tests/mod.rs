fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
