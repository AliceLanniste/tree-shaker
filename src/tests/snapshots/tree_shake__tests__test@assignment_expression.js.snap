---
source: src/tests/mod.rs
expression: tree_shake(&input)
input_file: src/tests/fixtures/assignment_expression.js
---
export function f1() {
	return 2;
}
export function f2(__unused_E8B5) {
	return 2;
}
export function f3(a, b) {
	if (b) a = 2;
	return a;
}
export function f4(__unused_7C4F, b) {
	return 3;
}
export function f5(a, b) {
	a = b;
	return a;
}
export function f6(a, b) {
	a.p = b;
	effect;
	return a;
}
